docs/enhancements/platform-integration.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Platform-Integration
+authors:
+  - "@vrutkovs"
+creation-date: 2022-07-26
+last-updated: 2022-07-26
+---
+
+# Platform integration
+
+## Summary
+Assisted installed has historically been used to configure baremetal clusters. Eventually, users have grown accustomed to use it for virtual machines in the cloud, as it helped them to validate networking settings. The installed cluster however could not make use of cloud features (like scaling, volumes or loadbalancers) as it was installed with platform: none. Later on basic platform integration was added, described in [Define Providers Interface](https://github.com/openshift/assisted-service/blob/master/docs/enhancements/add-external-providers.md).
+
+This enhancement extends providers API and describes how other components are interacting with this feature.
+
+## Motivation
+
+Adding other providers for integration with assisted installer should be easy and customizable, including various UX/UI changes for flexibility.
+
+### Goals
+
+- Define API changes necessary to support various providers
+- Define API for UI to allow being flexible about provider features
+- Define requirements for other components - most importantly, openshift-installer - required to be fulfilled to allow assisted-installer team add more integrations in the future.
+
```

Comment:
Is migrating existing platform integrations (metal, vsphere, nutanix) to use the new API a goal, or a non-goal?

Reply:
Good point, added as explicit goal